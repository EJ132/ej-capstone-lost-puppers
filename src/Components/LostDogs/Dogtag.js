import React, { Component } from 'react'
import '../../Stylesheets/DogTags.css'

export default class Dogtag extends Component {
    render(){
        return(
            <div className="dogTag">
                <h2>{this.props.name}</h2>
                <img src={this.props.img} alt={this.props.name}></img>
                <p>{this.props.category}</p>
                <p>{this.props.description}</p>
                <p>{this.props.dateCreated}</p>
            </div>
        )
    }
}