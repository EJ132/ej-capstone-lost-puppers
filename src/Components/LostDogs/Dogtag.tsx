/* eslint-disable eqeqeq */
import './DogTags.css'
import {Link} from 'react-router-dom'
import React from 'react'

function DogTag(props) {

    let date = new Date(props.dateCreated)

    return (
            <div className="dogTag">
                <Link to={`/find/${props.id}`}>
                    <h2>{props.name}</h2>
                    <img src={props.img} alt={props.name}></img>
                    <p>{props.category}</p>
                    <p>{props.description}</p>
                    <p>Posted: {date.getUTCFullYear()}/{(date.getUTCMonth() + 1)}/{date.getUTCDate()}</p>
                </Link>
            </div>
        )
}

export default DogTag;