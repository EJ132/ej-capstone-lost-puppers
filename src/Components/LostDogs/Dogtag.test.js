import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Dogtag from './Dogtag'

describe(`Dogtag function/component`, () => {

  const props = {
      id: 1,
      name: 'Rico', 
      img: 'http://google.com',
      category: 'Small',
      description: 'Test subject',
      date_created: '2029-01-22T08:00:00.000Z',
  }

  it('renders a Dogtag', () => {
    const wrapper = shallow(<Dogtag />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a div with with all dogTag info', () => {
    const div = shallow(<Dogtag {...props}/>)
      .find('.dogTag')
    expect(toJson(div)).toMatchSnapshot()
  })

})
