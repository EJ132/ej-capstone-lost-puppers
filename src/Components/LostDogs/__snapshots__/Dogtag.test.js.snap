// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dogtag function/component renders a Dogtag 1`] = `
<div
  className="dogTag"
>
  <Link
    to="/find/undefined"
  >
    <h2 />
    <img />
    <p />
    <p />
    <p>
      Posted: 
      NaN
      /
      NaN
      /
      NaN
    </p>
  </Link>
</div>
`;

exports[`Dogtag function/component renders a div with with all dogTag info 1`] = `
<div
  className="dogTag"
>
  <Link
    to="/find/1"
  >
    <h2>
      Rico
    </h2>
    <img
      alt="Rico"
      src="http://google.com"
    />
    <p>
      Small
    </p>
    <p>
      Test subject
    </p>
    <p>
      Posted: 
      NaN
      /
      NaN
      /
      NaN
    </p>
  </Link>
</div>
`;
